# Linter Rule: Disallow redundant use of `all` in `herb:disable` comments

**Rule:** `herb-disable-comment-no-redundant-all`

## Description

Prevents using `all` together with specific rule names in `<%#Â herb:disable ... %>` comments, as this is redundant.

## Rationale

When you use `all` in a `<%#Â herb:disable ... %>` comment, it disables every linter rule for that line. Adding specific rule names alongside `all` is redundant because `all` already covers them.

This rule helps maintain clean, concise disable comments by ensuring you either use `herb:disable all` by itself to disable all rules or lists only the specific rules you want to disable.

## Examples

### âœ… Good

```erb
<DIV>test</DIV> <%# herb:disable all %>

<DIV class='value'>test</DIV> <%# herb:disable html-tag-name-lowercase, html-attribute-double-quotes %>

<DIV>test</DIV> <%# herb:disable html-tag-name-lowercase %>
```

### ðŸš« Bad

```erb
<DIV>test</DIV> <%# herb:disable all, html-tag-name-lowercase %>

<DIV>test</DIV> <%# herb:disable html-tag-name-lowercase, all, html-attribute-double-quotes %>

<DIV>test</DIV> <%# herb:disable all, all %>
```

## Fix

Remove either the `all` keyword or the specific rule names:

**Option 1:** Keep only `all`
```erb
<DIV class='value'>test</DIV> <%# herb:disable all %>
```

**Option 2:** Keep only specific rules
```erb
<DIV class='value'>test</DIV> <%# herb:disable html-tag-name-lowercase, html-attribute-double-quotes %>
```

## References

\-
